DSEGOG-351 integrate favourite filters with filters

joshdimanteto opened the pull request:
## Description
- add favorite button
- add sections in autocomplete
- filter out duplicate filters  

## Testing instructions
Add a set up instructions describing how the reviewer should test the code

- [ ] Review code
- [ ] Check Actions build
- [ ] Review changes to test coverage
- [ ] {more steps here}

## Agile board tracking
closes DSEGOG-351

joshdimanteto committed:
DSEGOG-350 Delete favourite filter

joshdimanteto committed:
DSEGOG-351 integrate favourite filters with filters

joshdimanteto committed:
Merge branch 'DSEGOG-350-delete-favourite-filter' into DSEGOG-351-integrate-favourite-filters-with-filters

joshdimanteto committed:
Merge branch 'DSEGOG-350-delete-favourite-filter' into DSEGOG-351-integrate-favourite-filters-with-filters

louise-davies reviewed (changes requested):
Looks good overall, just some weirdness with the existing favourite filters... When I was filtering one of the favourite filters kept being duplicated. It seems as though the issue occurs when there's two _identical_ favourite filters (i.e the same tokens - occurs no matter whether they have the same name).

![image](https://github.com/user-attachments/assets/a9b36b3a-b889-4e93-946c-a45fef3eaa98)

Probably should either prevent duplicate favourite filters, (and at that point probably also disallow duplicate names as that seems like it could get confusing for users), or figure out a way to make the autocomplete happier with duplicates...

joshdimanteto commented:
I think the best approach would be to prevent duplicate favourite filters and avoid duplicate names, as that could get confusing for users. I can add error validation to the frontend, but it would be better if this validation is also done on the backend. That way, even users interacting purely through the API can't input bad data.

joshdimanteto committed:
Address review comments DSEG)G-351

- Add duplicate name check
- Change the key for favourite filter list item in the autocomplete to inculde the name

joshdimanteto committed:
Merge branch 'develop' into DSEGOG-351-integrate-favourite-filters-with-filters

louise-davies reviewed (approved):
LGTM - fixes the issue with duplicate filters being duplicated in the autocomplete and also prevents confusion between favourite filters with the same name :)
